### PR TITLE
support exact kind in OTLP metrics exporter

### PR DESCRIPTION
**Pull request by hstan**
When using `exactSelector` we found that the OTLP exporter does not handle array aggregator's kind (`ExactKind`).
This PR tries to fix it.

### CODE REVIEW

**Comment by hstan** [comment]
@jmacd seen from the failed test, I noticed it's explicitly asserted this shuold not be supported https://github.com/open-telemetry/opentelemetry-go/blob/master/exporters/otlp/internal/transform/metric_test.go#L349

I'm not sure why is that, does this change makes sense at all?

File: exporters/otlp/internal/transform/metric.go [hunk 1]
```diff
@@ -289,11 +289,74 @@ func Record(r export.Record) (*metricpb.Metric, error) {
...
+func gaugeArray(record export.Record, points []otel.Number) (*metricpb.Metric, error) {
+	desc := record.Descriptor()
+	m := &metricpb.Metric{
+		Name:        desc.Name(),
+		Description: desc.Description(),
+		Unit:        string(desc.Unit()),
+	}
+
+	start := record.StartTime()
+	end := record.EndTime()
```
Comment (MrAlias) [inline exporters/otlp/internal/transform/metric.go:318]: Probably worth declaring these variables as the return value of `toNanos` if they are both going to be used that way anyways.

```suggestion
	start := toNanos(record.StartTime())
	end := toNanos(record.EndTime())
```

File: exporters/otlp/internal/transform/metric.go [hunk 2]
```diff
@@ -289,11 +289,74 @@ func Record(r export.Record) (*metricpb.Metric, error) {
...
+		}
+
+		return gaugeArray(r, pts)
+
 	default:
 		return nil, fmt.Errorf("%w: %T", ErrUnimplementedAgg, agg)
 	}
 }
 
+func gaugeArray(record export.Record, points []otel.Number) (*metricpb.Metric, error) {
```
Comment (MrAlias) [inline exporters/otlp/internal/transform/metric.go:309]: Any reason to not just call this function `array`?

**Review by MrAlias — APPROVED** [review]

**Comment by MrAlias** [comment]
> @jmacd seen from the failed test, I noticed it's explicitly asserted this shuold not be supported https://github.com/open-telemetry/opentelemetry-go/blob/master/exporters/otlp/internal/transform/metric_test.go#L349
> 
> I'm not sure why is that, does this change makes sense at all?

The change seems to make sense to me. If you configure the metrics telemetry pipeline to send exact values they should not error the exporter. The tests will need to be updated as part of this PR.

https://github.com/open-telemetry/opentelemetry-go/blob/187adeb25163b4deb529b770649ec5d0afd21206/exporters/otlp/internal/transform/metric_test.go#L351-L353

File: exporters/otlp/internal/transform/metric.go (hunk 2 above)
Comment (hstan) [reply exporters/otlp/internal/transform/metric.go:309]: yeah I used `array` initially and the linter is not happy. There is a pacakge `go.opentelemetry.io/otel/sdk/metric/aggregator/array`.

File: exporters/otlp/internal/transform/metric.go (hunk 1 above)
Comment (hstan) [reply exporters/otlp/internal/transform/metric.go]: I inlined the variable, it makes more sense in the switch. Also aligns with other transform functions.

**Review by MrAlias — APPROVED** [review]
:+1:

**Comment by hstan** [comment]
@MrAlias 
I think the build failed due to a flaky test, I've no permission to rerun (passed locally)

**Comment by jmacd** [comment]
(Sorry all, I will catch up with this PR later today!!)

**Review by jmacd — APPROVED** [review]

**Comment by jmacd** [comment]
To be explicit, it is sensible and correct to expose exact values as Gauge data points.
We will eventually have a specification that states this behavior. :grinning:
Thanks 💪.

**Comment by jmacd** [comment]
I see a connection with https://github.com/open-telemetry/opentelemetry-proto/issues/188. There's no way to know that the resulting Gauge data points represent raw data, but I would like there to be.

**Review by Aneurysm9 — APPROVED** [review]

**Comment by Aneurysm9** [comment]
Apologies, this appears to be tripped up by a change to package structure, likely #1321.

**Pull request merged**